tests/expected/derive-arbitrary/safety_constraint_helper/safety_constraint_helper.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check that the `#[safety_constraint(...)]` attribute helper adds the conditions provided to
+//! the attribute to the derived `Arbitrary` implementation.
+
+extern crate kani;
+use kani::Invariant;
+
+#[derive(kani::Arbitrary)]
+struct PositivePoint {
+    #[safety_constraint(*x >= 0)]
```

Comment:
It is possible to use `self.x` if you're only deriving `Invariant`. That's because `self` doesn't mean anything in `any()`'s context, and we want to use the same condition in both. Otherwise, the error message is this:
```
error[E0424]: expected value, found module `self`
  --> src/main.rs:35:25
   |
32 | #[derive(kani::Arbitrary)]
   |          --------------- this function doesn't have a `self` parameter
...
35 |     #[safety_constraint(self.x == 0)]
   |                         ^^^^ `self` value is a keyword only available in methods with a `self` parameter

error: aborting due to 1 previous error
```

Reply:
BTW this has been updated in the PR description.